docs/agent_hocon_reference.md

```diff
@@ -0,0 +1,497 @@
+Each key's value is a dictionary describing the single argument value itself.
+This dictionary has the following keys:
+
+| Property Key | Description |
+|:-------------|:------------|
+| description  | A string which describes the particular argument |
+| type         | A string describing the type of the property. (See below) |
+| default      | An optional default value for the property |
+
+Scalar types here can be "int", "float", "string", "bool".
+It is possible that a properties' type can be "array"s for lists or "object"s for nested dictionaries.
+
+Any sample agent hocon with more than one agent will have some example of a simple properties dictionary.
+For a concrete, more complex properties definition, with nested objects and arrays,
+See the definition of [cao_item in the esp_descision_assistant.hocon](https://github.com/leaf-ai/neuro-san/blob/main/neuro_san/registries/esp_decision_assistant.hocon).
+
+##### required
+
+This is an optional list of string keys in the [properties](#properties) dictionary that are considered
+to be required whenever an upstream agent calls the one being described.
+Note that it's possible to specify a default value for any property that is not listed as required.
+
+### instructions
+
+When included, the single (often very long) string value here tells an LLM-enabled agent
+what it needs to do.
+
+This is optional because not every tool listed in the agent network uses an LLM (there exist CodedTools).
+But if you expect an agent to use an LLM, this instructions field is a must.
+
+### command
+
+An optional string to set an LLM-enabled agent in motion.
+
+### tools (agents)
+
+An optional list of strings with the names of other agents available for the agent being described to call.
+
+Typically the names listed here are other agents within the same agent network definition,
+often forming a tree structure, but overall agent networks are allowed to contain cycles.
+
+It is important to note that just because an agent is listed in the tools does not mean that it will always
+be called by an LLM.  The tools listing is merely a full description of what is _available_ to the agent.
+It is the agent itself that actually decides which tools (if any) to invoke depending on its instructions
+and the context of its query.
+
+#### External Agents
+
+This is not a hocon file key, but more a description of a concept that relates to listings of tools.
+
+It is possible for any agent to reference another agent on the same server by adding a forward-slash
+in front of the served agent's name.  This is typically the stem of an agent network hocon file in
+a deployment's registries directory.
+
+Example: "/website_search" or "/math_guy"
+
+This allows common agent network definitions to be used as functions for other local networks.
+
+Furthermore, it is also possible to reference agents on other neuro-san _servers_ by using a URL as a tool reference.
+
+Example: "http://localhost:8080/math_guy"
```

Comment:
What about `https`?

Reply:
Sure, it's a URL, but purposefully left this out.  This is a developer-centric doc and no developer is going to go jump through the hoops to get all the certs in line for a local https server.